### PR TITLE
[proxmox] update doc for lxc storage

### PR DESCRIPTION
**Pull request by xilmen**
##### SUMMARY
update doc for lxc storage

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/cloud/misc/proxmox.py

ADDITIONAL INFORMATION
https://github.com/ansible-collections/community.general/pull/4236

### CODE REVIEW

**Comment by felixfontein** [comment]
Ref: #4156

**Comment by felixfontein** [comment]
CC @thuttinpasseron @reitermarkus @m-rtijn

**Comment by thuttinpasseron** [comment]
IMHO instead of changing the examples to make it look like the storage option is necessary (which it isn't since if the "local" storage backend supports container volumes no error will be thrown), it would be better to better document the storage option with a clearer indication of what it means. Something along the lines of: If the `disk` option is not provided, the container volumes will be created on this storage backend, make sure that this storage backend supports container volumes.

**Comment by reitermarkus** [comment]
> if the "local" storage backend supports container volumes

When is this the case? With LVM, only `local-lvm` supports volumes and with ZFS, only `local-zfs` supports volumes, at least by default.

**Comment by thuttinpasseron** [comment]
> > if the "local" storage backend supports container volumes
> 
> When is this the case? With LVM, only `local-lvm` supports volumes and with ZFS, only `local-zfs` supports volumes, at least by default.

In my setup (I installed Debian then ProxMox VE in order to support a RAID 5 array but not in a ZFS pool), the "local" storage backend is a dir storage backend, and I can select Containers as a storage type, so I wouldn't need the `storage` option.

playbook:
```yaml
---
- name: Create a test CT with default options
  hosts: proxmox-host

  tasks:
  - name: Create test Container
    community.general.proxmox:
      api_host: "{{ hostvars[proxmox_host].ansible_host }}"
      api_token_id: "{{ proxmox_api_token_id }}"
      api_token_secret: "{{ proxmox_api_token }}"
      api_user: "{{ proxmox_api_user }}"
      vmid: 900
      node: "{{ proxmox_host }}"
      ostemplate: local:vztmpl/ubuntu-21.10-standard_21.10-1_amd64.tar.zst
      hostname: test.loc
      state: present
    delegate_to: localhost
```

output:
```
PLAY [Create a test CT with default options] ***********************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [proxmox-host]

TASK [Create test Container] *****************************************************************************************************************************************************************************************************************
changed: [proxmox-host -> localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
proxmox-host               : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

**Comment by reitermarkus** [comment]
> I can select Containers as a storage type

I see. Yes, I can select that too on my `local` storage, while it is already enabled by default for my `local-zfs` storage.

**Comment by m-rtijn** [comment]
> IMHO instead of changing the examples to make it look like the storage option is necessary (which it isn't since if the "local" storage backend supports container volumes no error will be thrown), it would be better to better document the storage option with a clearer indication of what it means. Something along the lines of: If the disk option is not provided, the container volumes will be created on this storage backend, make sure that this storage backend supports container volumes.

I agree that this is a better approach. This PR will only fix it for certain cases, and break it for others, as far as I can see.

**Comment by xilmen** [comment]
I have committed the changes by considering your recommendations

File: plugins/modules/cloud/misc/proxmox.py [hunk 1]
```diff
@@ -90,6 +90,7 @@
   storage:
     description:
       - target storage
+      - If the disk option is not provided, the container volumes will be created on this storage backend, make sure that this storage backend supports container volumes.
```
Comment (thuttinpasseron) [inline plugins/modules/cloud/misc/proxmox.py:93]:
```suggestion
      - If the C(disk) option is not provided, the container volumes will be created on this storage backend, make sure that this storage backend supports container volumes.
```

**Comment by felixfontein** [comment]
@reitermarkus @m-rtijn what do you think of the current version of the PR?

**Comment by felixfontein** [comment]
@reitermarkus @m-rtijn ping :)

File: plugins/modules/cloud/misc/proxmox.py [hunk 2]
```diff
@@ -214,6 +215,7 @@
     api_host: node1
     password: 123456
     hostname: example.org
+    storage: local-lvm
```
Comment (m-rtijn) [inline plugins/modules/cloud/misc/proxmox.py:218]: Is there a specific reason to still have the storage option in this example (same for line 312)?

Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox.py:218]: @xilmen is there a reason you marked this as resolved without a comment?

Comment (xilmen) [reply plugins/modules/cloud/misc/proxmox.py:218]: I add comment "Yes, In the first examples, we have specified "with minimal parameters", so for the following examples I have added it. This storage parameter blocking in some cases. Adding it, could help users to understand the errors they encounter."

**Review by xilmen — COMMENTED** [review]
Yes, In the first examples, we have specified "with minimal parameters", so for the following examples I have added it. This storage parameter blocking in some cases. Adding it, could help users to understand the errors they encounter.

File: plugins/modules/cloud/misc/proxmox.py (hunk 2 above)
Comment (xilmen) [reply plugins/modules/cloud/misc/proxmox.py:218]: But I had not published it, sorry.

**Comment by felixfontein** [comment]
@xilmen you might want to push your changes. You still haven't pushed something since end of February.

**Comment by xilmen** [comment]
> 

I don't have any other commits to make.

**Comment by felixfontein** [comment]
Ah, I misunderstood what you were saying in https://github.com/ansible-collections/community.general/pull/4236#discussion_r851628350 and the following comment. I was assuming you were referencing a change you made (but apparently did not push).

In any case, until (at least) one of the maintainers approves it, this will still stay open because I still don't know whether everything is resolved or not.

**Comment by xilmen** [comment]
This PR is too old, @felixfontein i close it ?

**Comment by felixfontein** [comment]
I cannot judge whether this should be merged or not, and I cannot judge whether this should be closed or not either.

**Pull request closed without merging**